src/category_theory/limits/shapes/regular_mono.lean

```diff
@@ -61,6 +62,32 @@ def regular_mono.lift' {W : C} (f : X ⟶ Y) [regular_mono f] (k : W ⟶ Y)
   {l : W ⟶ X // l ≫ f = k} :=
 fork.is_limit.lift' regular_mono.is_limit _ h
 
+/-- If `h` is a regular mono and `g` is a pullback of `g`, then `g` is a regular mono. -/
+def regular_of_is_pullback_of_regular {P Q R S : C} {f : P ⟶ Q} {g : P ⟶ R} {h : Q ⟶ S} {k : R ⟶ S}
+  [hr : regular_mono h] (comm : f ≫ h = g ≫ k) (t : is_limit (pullback_cone.mk _ _ comm)) :
+regular_mono g :=
+{ Z := hr.Z,
+  left := k ≫ hr.left,
+  right := k ≫ hr.right,
+  w := by rw [← reassoc_of comm, ← reassoc_of comm, hr.w],
+  is_limit :=
+  begin
+    apply fork.is_limit.mk' _ _,
+    intro s,
+    have l₁ : (fork.ι s ≫ k) ≫ regular_mono.left = (fork.ι s ≫ k) ≫ regular_mono.right,
+      rw [category.assoc, s.condition, category.assoc],
+    let l₂ : fork hr.left hr.right := fork.of_ι (s.ι ≫ k) l₁,
+    let p₂ : pullback_cone h k := pullback_cone.mk (hr.is_limit.lift l₂) s.ι (hr.is_limit.fac _ walking_parallel_pair.zero),
+    refine ⟨t.lift p₂, t.fac p₂ walking_cospan.right, _⟩,
+    intros m w,
+    have z : m ≫ g = t.lift p₂ ≫ g,
+    { erw w, exact (t.fac p₂ walking_cospan.right).symm },
```

Comment:
I don't think this needs changing, but I thought I'd mention it in case you're not aware: If you find making the cone and then fiddling around with `lift` and `fac` a bit tedious, there are the `lift'` variants which "bundle" a lift and its properties. Using them, you could rewrite this section:
```suggestion
    obtain ⟨l, hl⟩ := fork.is_limit.lift' hr.is_limit _ l₁,
    obtain ⟨p, hp₁, hp₂⟩ := pullback_cone.is_limit.lift' t _ _ hl,
    refine ⟨p, hp₂, _⟩,
    intros m w,
    have z : m ≫ g = p ≫ g := w.trans hp₂.symm,
```
which you may or may not find nicer than what you have.

Reply:
Ah now I see what the `lift'` variants are for. I like this!